src/webgpu/web_platform/external_texture/video.spec.ts

```diff
@@ -339,6 +338,112 @@ it will honor rotation metadata.
     });
   });
 
+g.test('importExternalTexture,sampleWithVideoFrameWithVisibleRectParam')
+  .desc(
+    `
+Tests that we can import VideoFrames and sample the correct sub-rectangle when visibleRect
+parameters are present.
+`
+  )
+  .params(u =>
+    u //
+      .combineWithParams(checkNonStandardIsZeroCopyIfAvailable())
+      .combineWithParams(kVideoExpectations)
+  )
+  .fn(async t => {
+    const videoElement = getVideoElement(t, t.params.videoName);
+
+    await startPlayingAndWaitForVideo(videoElement, async () => {
+      const source = await getVideoFrameFromVideoElement(t, videoElement);
+
+      // All tested videos are derived from an image showing yellow, red, blue or green in each
+      // quadrant. In this test we crop the video to each quadrant and check that desired color
+      // is sampled from each corner of the cropped image.
+      const srcVideoHeight = 240;
+      const srcVideoWidth = 320;
+      let cropParams = [
+        // Top left (yellow)
+        {
+          _subRect: { x: 0, y: 0, width: srcVideoWidth / 2, height: srcVideoHeight / 2 },
+          _color: t.params._yellowExpectation,
```

Comment:
these probably don't need underscores in key. just `subRect` and `color` is OK

Reply:
Done.